### PR TITLE
Horovod Estimator validate check serialize_data_creation

### LINKED ISSUE #2714: serialize_data_creation is not used in validate of Horovod Estimator
Need to make the Lock optional like train.

### PR DESCRIPTION
**Pull request by cyita**
Fix #2714

### CODE REVIEW

File: pyzoo/zoo/orca/learn/pytorch/torch_runner.py [hunk 1]
```diff
@@ -200,8 +200,11 @@ def validate(self, data_creator, num_steps=None, profile=False, info=None):
         self._toggle_profiling(profile=profile)
 
         with self.timers.record("validation"):
```
Comment (hkvision) [inline pyzoo/zoo/orca/learn/pytorch/torch_runner.py:202]: Shall we put the timer include data loading or just validation?

Comment (cyita) [reply pyzoo/zoo/orca/learn/pytorch/torch_runner.py:202]: Maybe just validation is better since the timer in train_epoch just includes validation.

Comment (hkvision) [reply pyzoo/zoo/orca/learn/pytorch/torch_runner.py:202]: But now the data loading time is also recorded here, shall we move it out?
For training timer, it only calculates training time excluding data time.

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4004/

**Pull request merged**